<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsLayoutWidgetBase</class>
 <widget class="QWidget" name="QgsLayoutWidgetBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>311</width>
    <height>357</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Composition</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QgsScrollArea" name="scrollArea">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="focusPolicy">
      <enum>Qt::WheelFocus</enum>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>-627</y>
        <width>297</width>
        <height>1063</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QgsCollapsibleGroupBoxBasic" name="groupBox">
         <property name="title">
          <string>General Settings</string>
         </property>
         <layout class="QGridLayout" name="gridLayout" columnstretch="0,1">
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Reference map</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QgsLayoutItemComboBox" name="mReferenceMapComboBox">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="toolTip">
             <string>Specifies the master map for this composition, which is used to georeference composer exports and for scale calculation for item styles.</string>
            </property>
            <property name="editable">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QgsCollapsibleGroupBoxBasic" name="mSnapToGridGroupCheckBox">
         <property name="focusPolicy">
          <enum>Qt::StrongFocus</enum>
         </property>
         <property name="title">
          <string>Guides and Grid</string>
         </property>
         <property name="checkable">
          <bool>false</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <property name="syncGroup" stdset="0">
          <string notr="true">composeritem</string>
         </property>
         <property name="collapsed" stdset="0">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_2" columnstretch="0,0">
          <item row="1" column="1">
           <widget class="QgsLayoutUnitsComboBox" name="mGridSpacingUnitsCombo"/>
          </item>
          <item row="3" column="0">
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QgsDoubleSpinBox" name="mOffsetXSpinBox">
              <property name="prefix">
               <string>x: </string>
              </property>
              <property name="suffix">
               <string/>
              </property>
              <property name="maximum">
               <double>9999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QgsDoubleSpinBox" name="mOffsetYSpinBox">
              <property name="prefix">
               <string>y: </string>
              </property>
              <property name="suffix">
               <string/>
              </property>
              <property name="maximum">
               <double>9999.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="3" column="1">
           <widget class="QgsLayoutUnitsComboBox" name="mGridOffsetUnitsComboBox"/>
          </item>
          <item row="1" column="0">
           <widget class="QgsDoubleSpinBox" name="mGridResolutionSpinBox">
            <property name="prefix">
             <string/>
            </property>
            <property name="minimum">
             <double>0.500000000000000</double>
            </property>
            <property name="maximum">
             <double>9999.000000000000000</double>
            </property>
            <property name="showClearButton" stdset="0">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="5" column="0" colspan="2">
           <widget class="QgsSpinBox" name="mSnapToleranceSpinBox">
            <property name="suffix">
             <string> px</string>
            </property>
            <property name="maximum">
             <number>200</number>
            </property>
            <property name="showClearButton" stdset="0">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Grid spacing</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Grid offset</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="2">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Snap tolerance</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QgsCollapsibleGroupBoxBasic" name="groupBox_3">
         <property name="title">
          <string>Export Settings</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="4" column="0" colspan="2">
           <widget class="QCheckBox" name="mGenerateWorldFileCheckBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>If checked, a separate world file which georeferences exported images will be created</string>
            </property>
            <property name="text">
             <string>Save world file</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Export resolution</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QgsSpinBox" name="mResolutionSpinBox">
            <property name="suffix">
             <string> dpi</string>
            </property>
            <property name="prefix">
             <string/>
            </property>
            <property name="maximum">
             <number>3000</number>
            </property>
            <property name="showClearButton" stdset="0">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QCheckBox" name="mRasterizeCheckBox">
            <property name="toolTip">
             <string>If checked, exports from this layout will be rasterized.</string>
            </property>
            <property name="text">
             <string>Print as raster</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QCheckBox" name="mForceVectorCheckBox">
            <property name="toolTip">
             <string>If checked, the layout will always be kept as vector objects when exported to a compatible format, even if the appearance of the resultant file does not match the layouts settings. If unchecked, some elements in the layout may be rasterized in order to keep their appearance intact.</string>
            </property>
            <property name="text">
             <string>Always export as vectors</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QgsCollapsibleGroupBoxBasic" name="groupBox_5">
         <property name="title">
          <string>Resize Layout to Content</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,1">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Margin units</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QgsLayoutUnitsComboBox" name="mMarginUnitsComboBox"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="0" column="1">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Top margin</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="0" column="3">
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="0" column="2">
             <widget class="QgsDoubleSpinBox" name="mTopMarginSpinBox">
              <property name="singleStep">
               <double>0.100000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0" colspan="4">
             <layout class="QHBoxLayout" name="horizontalLayout_7">
              <item>
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>Left</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QgsDoubleSpinBox" name="mLeftMarginSpinBox">
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>Right</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QgsDoubleSpinBox" name="mRightMarginSpinBox">
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="2" column="1">
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>Bottom</string>
              </property>
             </widget>
            </item>
            <item row="2" column="2">
             <widget class="QgsDoubleSpinBox" name="mBottomMarginSpinBox">
              <property name="singleStep">
               <double>0.100000000000000</double>
              </property>
             </widget>
            </item>
            <item row="3" column="1" colspan="2">
             <widget class="QPushButton" name="mResizePageButton">
              <property name="text">
               <string>Resize layout</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QgsCollapsibleGroupBox" name="groupBox_2">
         <property name="title">
          <string>Variables</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QgsVariableEditorWidget" name="mVariableEditor" native="true">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>200</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QgsCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgscollapsiblegroupbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsScrollArea</class>
   <extends>QScrollArea</extends>
   <header>qgsscrollarea.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsCollapsibleGroupBoxBasic</class>
   <extends>QGroupBox</extends>
   <header>qgscollapsiblegroupbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>qgsdoublespinbox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsLayoutUnitsComboBox</class>
   <extends>QComboBox</extends>
   <header>qgslayoutunitscombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsSpinBox</class>
   <extends>QSpinBox</extends>
   <header>qgsspinbox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsLayoutItemComboBox</class>
   <extends>QComboBox</extends>
   <header>qgslayoutitemcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsVariableEditorWidget</class>
   <extends>QWidget</extends>
   <header location="global">qgsvariableeditorwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>scrollArea</tabstop>
  <tabstop>mReferenceMapComboBox</tabstop>
  <tabstop>mSnapToGridGroupCheckBox</tabstop>
  <tabstop>mGridResolutionSpinBox</tabstop>
  <tabstop>mGridSpacingUnitsCombo</tabstop>
  <tabstop>mOffsetXSpinBox</tabstop>
  <tabstop>mOffsetYSpinBox</tabstop>
  <tabstop>mGridOffsetUnitsComboBox</tabstop>
  <tabstop>mSnapToleranceSpinBox</tabstop>
  <tabstop>mResolutionSpinBox</tabstop>
  <tabstop>mRasterizeCheckBox</tabstop>
  <tabstop>mForceVectorCheckBox</tabstop>
  <tabstop>mGenerateWorldFileCheckBox</tabstop>
  <tabstop>mMarginUnitsComboBox</tabstop>
  <tabstop>mTopMarginSpinBox</tabstop>
  <tabstop>mLeftMarginSpinBox</tabstop>
  <tabstop>mRightMarginSpinBox</tabstop>
  <tabstop>mBottomMarginSpinBox</tabstop>
  <tabstop>mResizePageButton</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
